I try running the tests but I always get this error:

BUILD FAILED
/Users/ustun/gitrepo/sockso/build.xml:364: com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: Communications link failure

Last packet sent to the server was 0 ms ago.
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:39)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
    at com.mysql.jdbc.Util.handleNewInstance(Util.java:406)
    at com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:1074)
    at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2103)
    at com.mysql.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:718)
    at com.mysql.jdbc.JDBC4Connection.<init>(JDBC4Connection.java:46)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:39)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:513)
    at com.mysql.jdbc.Util.handleNewInstance(Util.java:406)
    at com.mysql.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:302)
    at com.mysql.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:282)
    at org.apache.tools.ant.taskdefs.JDBCTask.getConnection(JDBCTask.java:370)
    at org.apache.tools.ant.taskdefs.SQLExec.getConnection(SQLExec.java:942)
    at org.apache.tools.ant.taskdefs.SQLExec.execute(SQLExec.java:614)
    at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
    at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
    at org.apache.tools.ant.Task.perform(Task.java:348)
    at org.apache.tools.ant.Target.execute(Target.java:390)
    at org.apache.tools.ant.Target.performTasks(Target.java:411)
    at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
    at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
    at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
    at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
    at org.apache.tools.ant.Main.runBuild(Main.java:809)
    at org.apache.tools.ant.Main.startAnt(Main.java:217)
    at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
    at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
Caused by: java.net.ConnectException: Connection refused
    at java.net.PlainSocketImpl.socketConnect(Native Method)
    at java.net.PlainSocketImpl.doConnect(PlainSocketImpl.java:351)
    at java.net.PlainSocketImpl.connectToAddress(PlainSocketImpl.java:213)
    at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:200)
    at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:432)
    at java.net.Socket.connect(Socket.java:529)
    at java.net.Socket.connect(Socket.java:478)
    at java.net.Socket.<init>(Socket.java:375)
    at java.net.Socket.<init>(Socket.java:218)
    at com.mysql.jdbc.StandardSocketFactory.connect(StandardSocketFactory.java:253)
    at com.mysql.jdbc.MysqlIO.<init>(MysqlIO.java:280)
    at com.mysql.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:2026)
    ... 28 more

What do you think I'm missing?

--- a/src/com/pugh/sockso/web/action/browse/Popularer.java
+++ b/src/com/pugh/sockso/web/action/browse/Popularer.java
@@ -90,7 +90,7 @@ public class Popularer extends WebAction {
                             " on al.id = t.album_id " +
                             " left outer join play_log l " +
                             " on l.track_id = t.id " +
-                        " group by artistId, artistName, albumId, albumName, trackId, " +
+                        " group by artistId, artistName, albumId, albumName, albumYear, trackId, " +
                             " trackName, trackPath, trackNo, dateAdded " +
                         " having count(l.id) > 0 " +
                         " order by count(l.id) desc " +

--- a/test/com/pugh/sockso/db/DatabaseTest.java
+++ b/test/com/pugh/sockso/db/DatabaseTest.java
@@ -73,7 +73,7 @@ public class DatabaseTest extends SocksoTestCase {
 
         doTableTest( "artists", new String[] { "id", "name", "date_added", "browse_name" }, db );
         
-        doTableTest( "albums", new String[] { "id", "artist_id", "name", "date_added" }, db );
+        doTableTest( "albums", new String[] { "id", "artist_id", "name", "year", "date_added" }, db );
 
         doTableTest( "tracks", new String[] { "id", "artist_id", "album_id", "name", "path", "length", "date_added", "collection_id", "track_no" }, db );
 

--- a/test/com/pugh/sockso/music/TrackTest.java
+++ b/test/com/pugh/sockso/music/TrackTest.java
@@ -93,6 +93,7 @@ public class TrackTest extends SocksoTestCase {
     public void testCreateVectorFromResultSet() throws SQLException {
 
         final String albumName = "my album name";
+        final String albumYear = "1984";
         
         // set up result set to return the info for 1 track
         final ResultSet rs = createMock( ResultSet.class );
@@ -101,6 +102,7 @@ public class TrackTest extends SocksoTestCase {
         expect( rs.getString("artistName") ).andReturn( albumName );
         expect( rs.getInt("albumId") ).andReturn( -1 );
         expect( rs.getString("albumName") ).andReturn( albumName );
+        expect( rs.getString("albumYear") ).andReturn( albumYear );
         expect( rs.getInt("trackId") ).andReturn( -1 );
         expect( rs.getString((String)anyObject()) ).andReturn( "1" ).times( 2 );
         expect( rs.getInt("trackNo") ).andReturn( 1 ).times( 1 );
@@ -114,6 +116,7 @@ public class TrackTest extends SocksoTestCase {
             assertEquals( 1, tracks.size() );
             final Track track = tracks.elementAt( 0 );
             assertEquals( albumName, track.getAlbum().getName() );
+            assertEquals( albumYear, track.getAlbum().getYear() );
             verify( rs );
         }
         

--- a/test/data/fixtures/emptyArtistsAndAlbums.fix
+++ b/test/data/fixtures/emptyArtistsAndAlbums.fix
@@ -1,4 +1,4 @@
 
 artists:1,My Artist,now(),My Artist
 
-albums:1,1,My Album,now()
+albums:1,1,My Album,1984,now()

--- a/test/data/fixtures/musicTree.fix
+++ b/test/data/fixtures/musicTree.fix
@@ -4,7 +4,7 @@ collection:1,/music
 artists:1,A Artist,now(),A Artist
 artists:2,Empty Artist,now(),Empty Artist
 
-albums:1,1,A Album,now()
-albums:2,1,Empty Album,now()
+albums:1,1,A Album,1984,now()
+albums:2,1,Empty Album,1984,now()
 
 tracks:1,1,1,My Track,/music/track.mp3,180,now(),1,1

--- a/test/data/fixtures/playlistTracks.fix
+++ b/test/data/fixtures/playlistTracks.fix
@@ -3,7 +3,7 @@ collection:1,/music
 
 artists:1,My Artist,now(),My Artist
 
-albums:1,1,My Album,now()
+albums:1,1,My Album,1984,now()
 
 tracks:1,1,1,My Track,/music/01 - track.mp3,180,now(),1,1
 tracks:2,1,1,Second Track,/music/subfolder/03 - can track.mp3,180,now(),1,1

--- a/test/data/fixtures/singleTrack.fix
+++ b/test/data/fixtures/singleTrack.fix
@@ -5,6 +5,6 @@ collection:1,/music
 
 artists:1,My Artist,now(),My Artist
 
-albums:1,1,My Album,now()
+albums:1,1,My Album,1984,now()
 
 tracks:1,1,1,My Track,/music/track.mp3,180,now(),1,1

--- a/test/data/fixtures/tracks.fix
+++ b/test/data/fixtures/tracks.fix
@@ -5,6 +5,6 @@ collection:1,/music
 
 artists:1,My Artist,now(),My Artist
 
-albums:1,1,My Album,now()
+albums:1,1,My Album,1984,now()
 
 tracks:1,1,1,My Track,/music/track.mp3,180,now(),1,1

--- a/test/data/fixtures/tracksForPath.fix
+++ b/test/data/fixtures/tracksForPath.fix
@@ -5,7 +5,7 @@ collection:1,/music
 
 artists:1,My Artist,now(),My Artist
 
-albums:1,1,My Album,now()
+albums:1,1,My Album,1984,now()
 
 tracks:1,1,1,My Track 2,/music/b_other/track.mp3,180,now(),1,1
 tracks:2,1,1,My Track 1,/music/a_track.mp3,180,now(),1,1

